x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRuleCriteria.java

```diff
@@ -23,18 +27,40 @@
 import org.elasticsearch.xcontent.XContentType;
 
 import java.io.IOException;
+import java.util.List;
 import java.util.Locale;
 import java.util.Objects;
 
 import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+import static org.elasticsearch.xpack.application.rules.QueryRuleCriteria.CriteriaType.GLOBAL;
 
 public class QueryRuleCriteria implements Writeable, ToXContentObject {
     private final CriteriaType criteriaType;
     private final String criteriaMetadata;
-    private final Object criteriaValue;
+    private final List<Object> criteriaValues;
+
+    private static final Logger logger = LogManager.getLogger(QueryRuleCriteria.class);
 
     public enum CriteriaType {
-        EXACT;
+        GLOBAL(true, true),
+        EXACT(true, true),
+        EXACT_FUZZY(true, false),
+        PREFIX(true, false),
+        SUFFIX(true, false),
+        CONTAINS(true, false),
+        LT(false, true),
+        LTE(false, true),
+        GT(false, true),
+        GTE(false, true);
+
+        final boolean supportsStringMatchInput;
+        final boolean supportsNumericMatchInput;
+
+        CriteriaType(boolean supportsStringMatchInput, boolean supportsNumericMatchInput) {
```

Comment:
How about adding a couple of Predicate functions for matching a string or a number? That way, the enums would do the match themselves instead of doing the switch on the criteria type...

Reply:
Yes, that's cleaner ☕

Reply:
Still feels a bit clunky to me, but WDYT?

Reply:
I like much better that the specific behaviour is now part of the individual enums. Yes, it's still not super clean, but it's not an easy one - going full class hierarchy is muddier, and using predicates for determining behaviour is not super ideal as well.

I think moving it to a separate file will help here as we encapsulate that logic.

Reply:
I was debating yesterday whether it should be moved to a separate file. It might be a little less clunky out of the way there.